yaml/hints.nim

```diff
@@ -26,13 +26,13 @@ type
     ## ================== =========================
     ## Name               RegEx
     ## ================== =========================
-    ## ``yTypeInteger``   ``0 | -? [1-9] [0-9]*``
-    ## ``yTypeFloat``     ``-? [1-9] ( \. [0-9]* [1-9] )? ( e [-+] [1-9] [0-9]* )?``
-    ## ``yTypeFloatInf``  ``-? \. (inf | Inf | INF)``
-    ## ``yTypeFloatNaN``  ``-? \. (nan | NaN | NAN)``
-    ## ``yTypeBoolTrue``  ``y|Y|yes|Yes|YES|true|True|TRUE|on|On|ON``
-    ## ``yTypeBoolFalse`` ``n|N|no|No|NO|false|False|FALSE|off|Off|OFF``
-    ## ``yTypeNull``      ``~ | null | Null | NULL``
+    ## ``yTypeInteger``   ``[-+]? [0-9]+``
+    ## ``yTypeFloat``     ``[-+]? ( \. [0-9]+ | [0-9]+ ( \. [0-9]* )? ) ( [eE] [-+]? [0-9]+ )?``
+    ## ``yTypeFloatInf``  ``[-+]? ( \.inf | \.Inf | \.INF )``
+    ## ``yTypeFloatNaN``  ``\.nan | \.NaN | \.NAN``
+    ## ``yTypeBoolTrue``  ``y|Y|yes|Yes|YES|true|True|TRUE|on|On|ON`` (and all upper/lower variants)
+    ## ``yTypeBoolFalse`` ``n|N|no|No|NO|false|False|FALSE|off|Off|OFF``  (and all upper/lower variants)
```

Comment:
Normally this would be out of scope for this PR but since you did include tests for boolean hints, I have to comment that the regex used here is YAML 1.1, which should have never been used, and when this code is touched, it should be updated to YAML 1.2, which is

    true | True | TRUE | false | False | FALSE

and nothing else.